doc/sphinx/language/core/variants.rst

```diff
@@ -64,6 +67,24 @@ be defined using :cmd:`Variant`.
       Variant option (A : Type) : Type := None : option A | Some : A -> option A.
       Variant sum (A B : Type) : Type := inl : A -> sum A B | inr : B -> sum A B.
 
+.. example:: :cmd:`Variant` won't define recursive types
+
+   .. rocqtop:: all
+
+      Fail Variant my_nat := zero | succ (n : my_nat).
+
+   The type :g:`my_nat` is said to be recursive because parameters of this type occur
```

Comment:
Saying "parameters" here sounds confusing. I would say "my_nat is recursive because it occurs in an argument of its constructors"